server/src/endpoints/store.rs

```diff
@@ -149,8 +181,19 @@ fn store_event(
                         })
                         .map_err(BadStoreRequest::ScheduleFailed)
                         .and_then(|result| result.map_err(BadStoreRequest::ProcessingFailed))
+                        .map_err(move |e| {
+                            outcome_producer.do_send(OutcomeMessage {
```

Comment:
_nit_: We have 4 calls to outcome producer in this chain of futures. Each of those calls happens before emitting an error. Would it be possible to move this all the way to the bottom so that there is only a single call?